### ISSUE TITLE
`tk_to_qiskit`: Support of `Qiskit.QuantumCircuit.Layout` Attribute

### ISSUE THREAD

**Issue by nquetschlich** [CQCL/pytket-qiskit#123]
Hello,

in Qiskit v0.43.0 (Link: https://qiskit.org/documentation/locale/de_DE/release_notes.html#:~:text=Added%20a%20new%20attribute%2C%20layout), the `layout` attribute is officially introduced that stores a `TranspileLayout` to preserve mapping information of a compiled and mapped quantum circuit (e.g., `initial_layout`, `input_qubit_mapping`, `final_layout`).

I was wondering whether it is planned that pytket-qiskit's `tk_to_qiskit` supports/sets this attribute when transforming a quantum circuit mapped using pytket to a qiskit circuit.

**Comment by CalMacCQ** [comment]
Hi,

Interesting, I wasn't aware of this feature in qiskit until you mentioned it.

So just to clarify this would allow a circuit to be mapped/routed in pytket and then the layout information is stored as stored as an attribute in the qiskit `QuantumCircuit` returned by `tk_to_qiskit` ?

Would you find this fetaure useful?

**Comment by nquetschlich** [comment]
> Hi,
> 
> Interesting, I wasn't aware of this feature in qiskit until you mentioned it.
> 
> So just to clarify this would allow a circuit to be mapped/routed in pytket and then the layout information is stored as stored as an attribute in the qiskit `QuantumCircuit` returned by `tk_to_qiskit` ?
> 
> Would you find this fetaure useful?

Yes, that would be helpful to know after the compilation which logical qubit was initially assigned to which physical qubit. And ideally, also for the final outcome after considering necessary SWAP gates introduced during compilation to match a given device's connectivity.
